### DISCUSSION TITLE
Native support for Prometheus-Operator CRDs (PodMonitor, ServiceMonitor) in the Istio Helm Chart

### DISCUSSION THREAD

**Discussion by obervinov** [istio/istio#51390] [Q&A]
Hello everyone!

I have a question and couldn't find an answer among the issues in this repository. As far as I understand, the current Istio Helm chart only supports metric collection through `prometheus.io/*` annotations on sidecar pods and istiod pods.

At the same time, the de facto standard for monitoring solutions in Kubernetes clusters is the [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack), which does not support `prometheus.io/*` annotations and instead uses CRDs like `PodMonitor` and `ServiceMonitor` for configuring metric collection.

Prometheus is one of the critical components for gaining full observability from Istio and visualizing it in Kiali. Are there any significant reasons why the Istio chart does not natively support rendering these CRDs, and instead supports the outdated method of collecting metrics from Istio-proxy and istiod through prometheus annotations?

Maybe I missed some information, and I would appreciate it if you could provide more details about the support for Prometheus-Operator in the Istio Helm chart.

**Comment by howardjohn** [comment]
I don't think Istio will emit any third party CRs (or even our own Istio CRs) as part of its install; We have tried to keep our install lean so that things can be composed more easily. We do offer examples of these in samples/addons/extras/prometheus-operator.yaml. These are not documented on https://istio.io/latest/docs/ops/integrations/prometheus/, though, so that could use some improvement

**Reply by zirain** [reply]
agree, we should keep our helm as clean as possible, only focus on kubernetes core APIs SGTM.
